src/simple_tracker_activity_recorder/simple_tracker_activity_recorder/rosbag_recorder_node.py

```diff
@@ -0,0 +1,121 @@
+        self.writer.write('tracking_state', serialize_message(msg_tracking_state), ns)
+        self.writer.write('detection', serialize_message(msg_detection_array), ns)
+        self.writer.write('trajectory', serialize_message(msg_trajectory_array), ns)
+        self.writer.write('prediction', serialize_message(msg_prediction_array), ns)
+
+        status_message = f"(Sky360) Tracker Status: trackable:{msg_tracking_state.trackable}, alive:{msg_tracking_state.alive}, started:{msg_tracking_state.started}, ended:{msg_tracking_state.ended}"
+        self.get_logger().info(f'{self.get_name()} - {status_message}.')
+
+  def setup_rosbag(self):
+
+    # setup the rosbag so that we can record into it
+    db_uri = f'sky360_recordings/{datetime.now().strftime("%Y_%m_%d-%H_%M_%S")}'
+    self.get_logger().info(f'Creating database {db_uri}.')
+
+    self.writer = rosbag2_py.SequentialWriter()
+    
+    storage_options = rosbag2_py._storage.StorageOptions(uri=db_uri, storage_id='sqlite3')
+    converter_options = rosbag2_py._storage.ConverterOptions('', '')
+    self.writer.open(storage_options, converter_options)
+
+    image_topic_info = rosbag2_py._storage.TopicMetadata(name='masked_frame', type='sensor_msgs/msg/Image', serialization_format='cdr')
+    tracking_state_topic_info = rosbag2_py._storage.TopicMetadata(name='tracking_state', type='simple_tracker_interfaces/msg/TrackingState', serialization_format='cdr')
+    detection_topic_info = rosbag2_py._storage.TopicMetadata(name='detection', type='vision_msgs/msg/Detection2DArray', serialization_format='cdr')
+    trajectory_topic_info = rosbag2_py._storage.TopicMetadata(name='trajectory', type='simple_tracker_interfaces/msg/TrackTrajectoryArray', serialization_format='cdr')
+    prediction_topic_info = rosbag2_py._storage.TopicMetadata(name='prediction', type='simple_tracker_interfaces/msg/TrackTrajectoryArray', serialization_format='cdr')
+
+    self.writer.create_topic(image_topic_info)
+    self.writer.create_topic(detection_topic_info)
+    self.writer.create_topic(tracking_state_topic_info)
+    self.writer.create_topic(trajectory_topic_info)
+    self.writer.create_topic(prediction_topic_info)
+
+  def config_list(self) -> List[str]:
+    return []
+
+  def validate_config(self) -> bool:
+    valid = True
+    return valid          
+
+  def on_config_loaded(self, init: bool):
+    if init:
+      self.msg_frame: Image = None
+      self.msg_tracking_state: TrackingState = None
+      self.br = CvBridge()
+
+def main(args=None):
+
+  rclpy.init(args=args)
+
+  subscriber_qos_profile = get_topic_subscriber_qos_profile()
+  publisher_qos_profile = get_topic_publisher_qos_profile()
+
+  node = RosbagRecorderNode(subscriber_qos_profile, publisher_qos_profile)
+
+  try:
+    rclpy.spin(node)
+  except (KeyboardInterrupt, ExternalShutdownException):
+      pass
+  finally:
+      rclpy.try_shutdown()
+      node.destroy_node()
+      #rclpy.rosshutdown()
```

Comment:
Is this call needed?

Reply:
@matt-ashman which one mate?

Reply:
I suspect it will be garbage collected just fine, all of the examples I have seen do this sort of thing so I did it also.

Reply:
I suspect it will eventually be garbage collected just fine however all the examples I have seen do something along these lines so I thought I would do it as well.